pinot-spi/src/main/java/org/apache/pinot/spi/data/FieldSpec.java

```diff
@@ -45,6 +45,7 @@
 @SuppressWarnings("unused")
 public abstract class FieldSpec implements Comparable<FieldSpec>, Serializable {
   public static final int DEFAULT_MAX_LENGTH = 512;
+  public static final boolean DEFAULT_NULLABILITY = false;
```

Comment:
So right now a lot of people must be using the table level `enableNullHandling: true` config which creates null vectors in the segments.

Since those tables wouldn't have the field level nullability set, I guess we'll set their nullability as false?

I think when we set `enableNullHandling: true` at table-level, it effectively makes all the columns nullable, since:

1. You can do queries such as IS NULL/IS NOT NULL and they'll work correctly (for v1 engine).
2. If you add the queryOption `enableNullHandling=true`, then Pinot will also return actual nulls instead of the default null value. e.g. select num_orders from table where num_orders is null will returns rows which look like: `[null, null, ...]`.

Reply:
that's the idea. 
and we will gradually deprecate the table level nullability and switch to use column level. in which case we need to 
1. migate the schema to have nullable column setting
2. turn off table level enable nullablity. 

there's a whole lot of problem.
1. what happen to all the previously null setup columns in previous segments?
2. how do we handle column become nullable but some segments generated previously doesn't have the vector
3. I can think of a lot of other corner cases. 

This PR doesn't address any of these. It only give a more fine-grain configuration telling the query engine some columns are nullable while others are not --> in contrast to a table level, where query engine HAVE TO assume all columns are nullable, and generate less efficient plan

Reply:
CC @61yao who is expert in this ^